pkg/controller/resource/app/spec.go

```diff
@@ -0,0 +1,23 @@
+package app
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+)
+
+type StateGetter interface {
```

Comment:
I don't think it's the best name. Suggestions welcome.

Reply:
What about just `Interface`? We do that in `helmclient` and a few other places.

Reply:
Interface is ok, but should I name field `Resource.interface` then? Is it even allowed?

Reply:
Ack yes Resource.interface isn't nice even if it worked. What about `StateInterface`? 

Its the Getter part of the name I'm not keen on because I don't think we use it in the rest of our codebase.

Another thought. Should the Name be part of the interface or just a setting?

Reply:
Checked `Resource.interface` doesn't work. I think `StateGetter` goes more into conventions like `Reader` and `Closer`. Especially when `Name` is moved as a resource setting which I think is a good shout.

Reply:
👍 Cool, `StateGetter` makes sense to me.